### PR TITLE
added customChar functionality HOPEFULLY

### PR DESCRIPTION
**Pull request by tkun3**
Still need to test tomorrow morning but I think it'll work.....

### CODE REVIEW

**Comment by trstephen** [comment]
Trip report: It displayed `@` instead of `Hz` 🤔

edit: this makes sense, because [this line](https://github.com/trstephen/ceng355/pull/23/commits/5e964277cedf2a001244c390890770e8b2993909#diff-b8a659453e6ce6211a86c6f1d9486b58R111) loads the character at `0x40` which is `@`

File: src/lcd.c [hunk 1]
```diff
@@ -100,13 +99,16 @@ void LCD_Init(void){
...
+
     // Write the initial units and resistance / freq placeholders manually
     //   "  ???  H"
     //   "  ???  Ω"
     LCD_MoveCursor(LCD_FREQ_ROW, 3);
     LCD_SendText("???");
     LCD_MoveCursor(LCD_FREQ_ROW, 8);
-    LCD_SendASCIIChar("H");
+    //LCD_SendASCIIChar("H");
+    LCD_SendWord(LCD_DATA, 0x40);
```
Comment (trstephen) [inline src/lcd.c]: Whenever possible, avoid magic numbers. The location of 'Hz' should be a constant to 1) increase readability, and 2) make the connection explicit between this line and L156.

File: src/lcd.c [hunk 2]
```diff
@@ -100,13 +99,16 @@ void LCD_Init(void){
...
+    LCD_CustomCharInit();
+
     // Write the initial units and resistance / freq placeholders manually
     //   "  ???  H"
     //   "  ???  Ω"
     LCD_MoveCursor(LCD_FREQ_ROW, 3);
     LCD_SendText("???");
     LCD_MoveCursor(LCD_FREQ_ROW, 8);
-    LCD_SendASCIIChar("H");
+    //LCD_SendASCIIChar("H");
```
Comment (trstephen) [inline src/lcd.c]: Delete instead of commenting out.

File: src/lcd.c [hunk 3]
```diff
@@ -137,6 +139,33 @@ void myGPIOB_Init(void){
     GPIO_Init(GPIOB, &GPIO_InitStruct);
 }
 
+void LCD_CustomCharInit(void){
+  char customChar[8] = {
+	  0b10001,
+    0b11111,
```
Comment (trstephen) [inline src/lcd.c:145]: Nit: align spacing

File: src/lcd.c [hunk 4]
```diff
@@ -137,6 +139,33 @@ void myGPIOB_Init(void){
     GPIO_Init(GPIOB, &GPIO_InitStruct);
 }
 
+void LCD_CustomCharInit(void){
+  char customChar[8] = {
```
Comment (trstephen) [inline src/lcd.c:143]: Nit: name should be more descriptive e.g. `hertzSymbol`

File: src/lcd.c [hunk 5]
```diff
@@ -137,6 +139,33 @@ void myGPIOB_Init(void){
...
+	  0b00100,
+	  0b01000,
+	  0b01110
+  };
+  // resets the RS and R/W pins
+  LCD_SendWord(LCD_COMMAND, 0x00);
+  // Sets CG RAM address
+  LCD_SendWord(LCD_COMMAND, 0x40);
+  // all subsequent data is written and auto incremented to CG RAM address
+  int a;
```
Comment (trstephen) [inline src/lcd.c]: Initialize this variable in your `for` loop so its scope is confined to the loop.

**Review by trstephen — CHANGES_REQUESTED** [review]
This is really close to working! I followed [this procedure](http://www.circuitvalley.com/2012/02/lcd-custom-character-hd44780-16x2.html) and found the bugs 😉

**Comment by tkun3** [comment]
Fixed the issues addressed, realized i was accessing wrong address for CGRAM like you showed in the doc

**Pull request closed without merging**